packages/jest-cli/src/search_source.js

```diff
@@ -238,4 +239,30 @@ export default class SearchSource {
       return Promise.resolve({tests: []});
     }
   }
+
+  async getTestPaths(
+    globalConfig: GlobalConfig,
+    changedFilesPromise: ?ChangedFilesPromise,
+  ): Promise<SearchResult> {
+    const searchResult = await this._getTestPaths(
+      globalConfig,
+      changedFilesPromise,
+    );
+
+    if (globalConfig.filter) {
+      let tests = searchResult.tests;
+
+      // $FlowFixMe: dynamic require.
+      const filter = require(globalConfig.filter);
+      const filtered = await filter(tests.map(test => test.path));
+
+      if (Array.isArray(filtered)) {
```

Comment:
Can we change this to throw with a descriptive error message like `The filter {path} you provided does not return a an array of tests blah blah` – I don't think silently ignoring the result is a good idea.